configure.ac

```diff
@@ -168,6 +168,29 @@ AC_CHECK_FUNCS([bzero gettimeofday memmove memset mkdir strdup strrchr \
                 socket strerror getifaddrs getloadavg getpwnam_r getpwuid_r \
                 getgrnam_r getgrgid_r secure_getenv])
 
+AC_MSG_CHECKING(for readdir_r deprecation)
+old_cxxflags=$CXXFLAGS
+CXXFLAGS="-Werror=deprecated-declarations"
```

Comment:
I think you want to leave the ${CXXFLAGS} variable in the new CXXFLAGS, in case the user has put anything weird in there to make it work on their machine.

From looking at your changeset, it may just be that fixing arg would have solved the problem. Although I suspect this is perhaps one case where we don't necessarily want to fail if the test fails for any reason, hence just testing for deprecated-declarations causing an error.